bastionzero/zli
# Attaching to existing open zli connections [CWC-667]

## Description of the change

This adds functionality to manage zli open connections. This includes three new zli commands `list-connections` which lists any open zli connections, `attach` which lets the user re-attach to an existing open zli connection and `close` which lets the user close an existing open zli connection.

Please make sure you test this against both key-splitting enabled and key-splitting disabled agents, since the flow is different there might exist bugs on the one that do not exist on the other.

To test this:
- Open a new connection to a target with zli
- Type `exit` to exit that connection. Note that this exit does not close anymore the connection
- Using `lc` (list-connections`) get a list of any open zli connections
- Using `attach` reconnect to any of those. They should work as you never exited
- "Stress test" it by attaching to the same connection by different zli/webapp clients

## Type of change


## Related JIRA tickets

Relates to JIRA: CWC-667

## Potential Security Impacts

For non-keysplitting agents, attaching to an existing connection will not remove any other attached clients. This is not a real threat but it should still be noted.

## Checklists

### Development
